File: Monolingual/PreferencesController.m
//
//  PreferencesController.m
//  Monolingual
//
//  Created by Ingmar Stein on Mon Apr 19 2004.
//  Copyright (c) 2004-2012 Ingmar Stein. All rights reserved.
//

#import "PreferencesController.h"

@implementation PreferencesController

- (void) awakeFromNib
{
	[[self window] setFrameAutosaveName:@"PreferencesWindow"];
}

- (IBAction) add: (id)sender
{
    NSOpenPanel *oPanel = [NSOpenPanel openPanel];

    [oPanel setAllowsMultipleSelection:YES];
	[oPanel setCanChooseDirectories:YES];
	[oPanel setCanChooseFiles:NO];
	[oPanel setTreatsFilePackagesAsDirectories:YES];

	// Weak references to NSWindowControllers are not supported on 10.7
	//__weak __typeof__(self) wself = self;
	__unsafe_unretained __typeof__(self) wself = self;
	[oPanel beginWithCompletionHandler:^(NSInteger result) {
		if (NSOKButton == result) {
			for (NSURL *url in [oPanel URLs]) {
				[wself.roots addObject:@{ @"Path" : [url path],
										  @"Languages" : @YES,
										  @"Architectures" : @YES}];
			}
		}
	}];
}

@end

File: Monolingual/PreferencesController.swift
//
//  PreferencesController.swift
//  Monolingual
//
//  Created by Ingmar Stein on Mon Apr 19 2004.
//  Copyright (c) 2004-2014 Ingmar Stein. All rights reserved.
//

import Cocoa

class PreferencesController : NSWindowController {

	@IBOutlet var roots: NSArrayController

	override func awakeFromNib() {
		self.window.setFrameAutosaveName("PreferencesWindow")
	}

	@IBAction func add(sender: AnyObject) {
		let oPanel = NSOpenPanel()

		oPanel.allowsMultipleSelection = true
		oPanel.canChooseDirectories = true
		oPanel.canChooseFiles = false
		oPanel.treatsFilePackagesAsDirectories = true

		oPanel.beginWithCompletionHandler {
			(result: NSInteger) in
			if NSOKButton == result {
				for obj in oPanel.URLs {
					let url = obj as NSURL
					self.roots.addObject([ "Path" : url.path,
										   "Languages" : true,
										   "Architectures" : true ])
				}
			}
		}
	}
}
